specs/core/0_beacon-chain.md

```diff
@@ -1474,12 +1455,9 @@ Below are the processing steps that happen at every `block`.
 
 ### RANDAO
 
-* Let `repeat_hash(x, n) = x if n == 0 else repeat_hash(hash(x), n-1)`.
 * Let `proposer = state.validator_registry[get_beacon_proposer_index(state, state.slot)]`.
-* Verify that `repeat_hash(block.randao_reveal, proposer.randao_layers) == proposer.randao_commitment`.
+* Verify that `bls_verify(pubkey=proposer.pubkey, message=proposer.randao_layers, signature=block.randao_reveal, domain=get_domain(state.fork, state.slot, DOMAIN_RANDAO))`.
 * Set `state.latest_randao_mixes[state.slot % LATEST_RANDAO_MIXES_LENGTH] = hash(xor(state.latest_randao_mixes[state.slot % LATEST_RANDAO_MIXES_LENGTH], block.randao_reveal))`
```

Comment:
Need to truncate the `randao_reveal` to 32 bytes by either doing `randao_reveal[0:32]` or `hash(randao_reveal)`. The former is cheaper and sufficient

Reply:
I think this is fine :) The `randao_reveal` is a signature.

Reply:
And is thus 96 bytes. Do we want the `latest_randao_mixes` to store 96 bytes per item rather than a `bytes32`?
`bytes32` is sufficient.

If we want `bytes96`, we need to fix the initialization of this array and make sure that the usage in shuffling of a 96 byte seed is fine. I'd lean toward truncating

Reply:
Oh I see this comment is about line 1460 (as opposed to line 1459). Still, `state.latest_randao_mixes` is storing a 32-byte `hash`, no? (As a side note, `randao_reveal[0:32]` isn't great because the top bits are flags which are not random.)

Reply:
There's still a 32 byte thing being xored with a 96 byte thing. Personally I would just remove the xor and do `state.latest_randao_mixes[state.slot % LATEST_RANDAO_MIXES_LENGTH] = hash(state.latest_randao_mixes[state.slot % LATEST_RANDAO_MIXES_LENGTH] + block.randao_reveal)`

Reply:
`hash` of the concatenation looks good.
Otherwise take the last 32 bytes (instead of the first with the flags) and xor.